### ISSUE TITLE
Permissions are wrong in 0.8.17-1 package

### ISSUE THREAD

**Issue by jbsbbn** [opennetworkinglab/flowvisor#199]
In the FV 0.8.17-1 package, the permissions are wrong on /usr/share/db/flowvisor and /etc/flowvisor:

```
+$ ls -Fla /usr/share/db/flowvisor
total 8
drwxr-xr-x 2 root root 4096 Apr 12 16:45 ./
drwxr-xr-x 3 root root 4096 Apr 12 16:45 ../

+$ ls -Fla /etc/flowvisor
total 28
drwxr-xr-x  2 root root 4096 Apr 12 16:45 ./
drwxr-xr-x 56 root root 4096 Apr 12 16:47 ../
-rw-r--r--  1 root root 4682 Feb 13 20:55 FlowVisorDB.sql
-rw-r--r--  1 root root   98 Feb 13 20:55 derby.properties
-rw-r--r--  1 root root 1952 Feb 13 20:55 envs.sh
-rw-r--r--  1 root root  808 Feb 13 20:55 fvlog.config
```

I'm pretty sure this wasn't a problem in the previous 0.8.17 package in the old repo. What broke?

This is a blocker bug, because you can't run 'sudo -u flowvisor fvconfig generate /etc/flowvisor/config.json' given those permissions (you get permission erors and a 275-line Java stacktrace, which I can send along if you really want it).

Can you fix this in a 0.8.17-2 package? We'd really prefer there not be a 0.8.18 release at this point, and changing the package version number seems like the right thing to do here, since this is purely a packaging issue, not a code bug.

I haven't yet checked 1.0.1, but will in a moment (and I'll make a new ticket for that if necessary).

**Comment by jbsbbn** [comment]
This doesn't seem to be a problem in the 1.0.1-1 package, so I haven't made a new ticket. (But wanted to mention somewhere that I checked.)

**Comment by leenheer** [comment]
Fixed in 0.8.17-2

**Comment by jbsbbn** [comment]
Did you test this? it still looks wrong to me, when I install it on a clean Debian 6.0 system:

```
wget http://updates.onlab.us/GPG-KEY-ONLAB
sudo apt-key add GPG-KEY-ONLAB
sudo sh -c 'echo "deb http://updates.onlab.us/debian stable/" >> /etc/apt/sources.list'
sudo apt-get update
sudo apt-get install -y flowvisor=0.8.17-2
```

That all worked fine, but the permissions still look wrong:

```
+$ ls -Fla /usr/share/db/flowvisor /etc/flowvisor
/etc/flowvisor:
total 28
drwxr-xr-x  2 root root 4096 Apr 22 10:57 ./
drwxr-xr-x 56 root root 4096 Apr 22 10:59 ../
-rw-r--r--  1 root root 4682 Apr 19 19:17 FlowVisorDB.sql
-rw-r--r--  1 root root   98 Apr 19 19:17 derby.properties
-rw-r--r--  1 root root 1952 Apr 19 19:17 envs.sh
-rw-r--r--  1 root root  808 Apr 19 19:17 fvlog.config

/usr/share/db/flowvisor:
total 8
drwxr-xr-x 2 root root 4096 Apr 22 10:57 ./
drwxr-xr-x 3 root root 4096 Apr 22 10:57 ../
```

And, of course, 'sudo -u flowvisor fvconfig generate /etc/flowvisor/config.json' still fails.

**Comment by leenheer** [comment]
Your system doesn't look clean to me (see Apr 19 date).

**Comment by jbsbbn** [comment]
Well, I provisioned a new ExoGENI VM for this, and I'm pretty sure it didn't have FV installed before, but I'll try again now. (I'm also not sure what the datestamp on /etc/flowvisor has to do with anything -- if the package creates that, it'll presumably have whatever date the package thinks it should have, right?)

In any case, I created a new one, confirmed that it didn't have either of those directories, installed FV 0.8.17-2, and confirmed that it got the directories with the wrong permissions. Here's the full output:

```
[11:19:04] jbs@carlin:/home/jbs
+$ ls -Fla /usr/share/db/flowvisor /etc/flowvisor
ls: cannot access /usr/share/db/flowvisor: No such file or directory
ls: cannot access /etc/flowvisor: No such file or directory

[11:19:04] jbs@carlin:/home/jbs
+$ sudo apt-get install -y flowvisor=0.8.17-2
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following extra packages will be installed:
  ca-certificates-java dbus flowvisor java-common libavahi-client3
  libavahi-common-data libavahi-common3 libcups2 libdbus-1-3 libfreetype6
  libjpeg62 liblcms1 libnspr4-0d libnss3-1d openjdk-6-jre-headless
  openjdk-6-jre-lib tzdata-java
Suggested packages:
  dbus-x11 default-jre equivs cups-common liblcms-utils libnss-mdns
  sun-java6-fonts ttf-dejavu-extra ttf-baekmuk ttf-unfonts ttf-unfonts-core
  ttf-sazanami-gothic ttf-kochi-gothic ttf-sazanami-mincho ttf-kochi-mincho
  ttf-wqy-microhei ttf-wqy-zenhei ttf-indic-fonts
The following NEW packages will be installed:
  ca-certificates-java dbus flowvisor java-common libavahi-client3
  libavahi-common-data libavahi-common3 libcups2 libdbus-1-3 libfreetype6
  libjpeg62 liblcms1 libnspr4-0d libnss3-1d openjdk-6-jre-headless
  openjdk-6-jre-lib tzdata-java
0 upgraded, 17 newly installed, 0 to remove and 0 not upgraded.
Need to get 41.7 MB of archives.
After this operation, 103 MB of additional disk space will be used.
Get:1 http://ftp.us.debian.org/debian/ squeeze/main libdbus-1-3 amd64 1.2.24-4+squeeze2 [146 kB]
Get:2 http://ftp.us.debian.org/debian/ squeeze/main dbus amd64 1.2.24-4+squeeze2 [236 kB]
Get:3 http://ftp.us.debian.org/debian/ squeeze/main openjdk-6-jre-lib all 6b18-1.8.13-0+squeeze2 [5955 kB]
Get:4 http://updates.onlab.us/debian/ stable/ flowvisor 0.8.17-2 [7245 kB]
Get:5 http://ftp.us.debian.org/debian/ squeeze/main ca-certificates-java all 20100412 [101 kB]
Get:6 http://ftp.us.debian.org/debian/ squeeze/main tzdata-java all 2012g-0squeeze1 [143 kB]
Get:7 http://ftp.us.debian.org/debian/ squeeze/main java-common all 0.40 [64.3 kB]
Get:8 http://ftp.us.debian.org/debian/ squeeze/main libavahi-common-data amd64 0.6.27-2+squeeze1 [113 kB]
Get:9 http://ftp.us.debian.org/debian/ squeeze/main libavahi-common3 amd64 0.6.27-2+squeeze1 [52.7 kB]
Get:10 http://ftp.us.debian.org/debian/ squeeze/main libavahi-client3 amd64 0.6.27-2+squeeze1 [57.2 kB]
Get:11 http://ftp.us.debian.org/debian/ squeeze/main libcups2 amd64 1.4.4-7+squeeze3 [234 kB]
Get:12 http://ftp.us.debian.org/debian/ squeeze/main liblcms1 amd64 1.18.dfsg-1.2+b3 [112 kB]
Get:13 http://ftp.us.debian.org/debian/ squeeze/main libjpeg62 amd64 6b1-1 [94.0 kB]
Get:14 http://ftp.us.debian.org/debian/ squeeze/main libnspr4-0d amd64 4.8.6-1 [139 kB]
Get:15 http://ftp.us.debian.org/debian/ squeeze/main libnss3-1d amd64 3.12.8-1+squeeze6 [1116 kB]
Get:16 http://ftp.us.debian.org/debian/ squeeze/main libfreetype6 amd64 2.4.2-2.1+squeeze4 [377 kB]
Get:17 http://ftp.us.debian.org/debian/ squeeze/main openjdk-6-jre-headless amd64 6b18-1.8.13-0+squeeze2 [25.5 MB]
Fetched 41.7 MB in 10s (3855 kB/s)                                             
Selecting previously deselected package libdbus-1-3.
(Reading database ... 12998 files and directories currently installed.)
Unpacking libdbus-1-3 (from .../libdbus-1-3_1.2.24-4+squeeze2_amd64.deb) ...
Selecting previously deselected package dbus.
Unpacking dbus (from .../dbus_1.2.24-4+squeeze2_amd64.deb) ...
Selecting previously deselected package openjdk-6-jre-lib.
Unpacking openjdk-6-jre-lib (from .../openjdk-6-jre-lib_6b18-1.8.13-0+squeeze2_all.deb) ...
Selecting previously deselected package ca-certificates-java.
Unpacking ca-certificates-java (from .../ca-certificates-java_20100412_all.deb) ...
Selecting previously deselected package tzdata-java.
Unpacking tzdata-java (from .../tzdata-java_2012g-0squeeze1_all.deb) ...
Selecting previously deselected package java-common.
Unpacking java-common (from .../java-common_0.40_all.deb) ...
Selecting previously deselected package libavahi-common-data.
Unpacking libavahi-common-data (from .../libavahi-common-data_0.6.27-2+squeeze1_amd64.deb) ...
Selecting previously deselected package libavahi-common3.
Unpacking libavahi-common3 (from .../libavahi-common3_0.6.27-2+squeeze1_amd64.deb) ...
Selecting previously deselected package libavahi-client3.
Unpacking libavahi-client3 (from .../libavahi-client3_0.6.27-2+squeeze1_amd64.deb) ...
Selecting previously deselected package libcups2.
Unpacking libcups2 (from .../libcups2_1.4.4-7+squeeze3_amd64.deb) ...
Selecting previously deselected package liblcms1.
Unpacking liblcms1 (from .../liblcms1_1.18.dfsg-1.2+b3_amd64.deb) ...
Selecting previously deselected package libjpeg62.
Unpacking libjpeg62 (from .../libjpeg62_6b1-1_amd64.deb) ...
Selecting previously deselected package libnspr4-0d.
Unpacking libnspr4-0d (from .../libnspr4-0d_4.8.6-1_amd64.deb) ...
Selecting previously deselected package libnss3-1d.
Unpacking libnss3-1d (from .../libnss3-1d_3.12.8-1+squeeze6_amd64.deb) ...
Selecting previously deselected package libfreetype6.
Unpacking libfreetype6 (from .../libfreetype6_2.4.2-2.1+squeeze4_amd64.deb) ...
Selecting previously deselected package openjdk-6-jre-headless.
Unpacking openjdk-6-jre-headless (from .../openjdk-6-jre-headless_6b18-1.8.13-0+squeeze2_amd64.deb) ...
Selecting previously deselected package flowvisor.
Unpacking flowvisor (from .../flowvisor_0.8.17-2_all.deb) ...
Creating FlowVisor (flowvisor) group.
Creating FlowVisor (flowvisor) user.
Processing triggers for man-db ...
Setting up libdbus-1-3 (1.2.24-4+squeeze2) ...
Setting up dbus (1.2.24-4+squeeze2) ...
Starting system message bus: dbus.
Setting up tzdata-java (2012g-0squeeze1) ...
Setting up java-common (0.40) ...
Setting up libavahi-common-data (0.6.27-2+squeeze1) ...
Setting up libavahi-common3 (0.6.27-2+squeeze1) ...
Setting up libavahi-client3 (0.6.27-2+squeeze1) ...
Setting up libcups2 (1.4.4-7+squeeze3) ...
Setting up liblcms1 (1.18.dfsg-1.2+b3) ...
Setting up libjpeg62 (6b1-1) ...
Setting up libnspr4-0d (4.8.6-1) ...
Setting up libnss3-1d (3.12.8-1+squeeze6) ...
Setting up libfreetype6 (2.4.2-2.1+squeeze4) ...
Setting up openjdk-6-jre-headless (6b18-1.8.13-0+squeeze2) ...
update-alternatives: using /usr/lib/jvm/java-6-openjdk/jre/bin/java to provide /usr/bin/java (java) in auto mode.
update-alternatives: using /usr/lib/jvm/java-6-openjdk/jre/bin/keytool to provide /usr/bin/keytool (keytool) in auto mode.
update-alternatives: using /usr/lib/jvm/java-6-openjdk/jre/bin/pack200 to provide /usr/bin/pack200 (pack200) in auto mode.
update-alternatives: using /usr/lib/jvm/java-6-openjdk/jre/bin/rmid to provide /usr/bin/rmid (rmid) in auto mode.
update-alternatives: using /usr/lib/jvm/java-6-openjdk/jre/bin/rmiregistry to provide /usr/bin/rmiregistry (rmiregistry) in auto mode.
update-alternatives: using /usr/lib/jvm/java-6-openjdk/jre/bin/unpack200 to provide /usr/bin/unpack200 (unpack200) in auto mode.
update-alternatives: using /usr/lib/jvm/java-6-openjdk/jre/bin/orbd to provide /usr/bin/orbd (orbd) in auto mode.
update-alternatives: using /usr/lib/jvm/java-6-openjdk/jre/bin/servertool to provide /usr/bin/servertool (servertool) in auto mode.
update-alternatives: using /usr/lib/jvm/java-6-openjdk/jre/bin/tnameserv to provide /usr/bin/tnameserv (tnameserv) in auto mode.
update-alternatives: using /usr/lib/jvm/java-6-openjdk/jre/lib/jexec to provide /usr/bin/jexec (jexec) in auto mode.
Setting up openjdk-6-jre-lib (6b18-1.8.13-0+squeeze2) ...
Setting up flowvisor (0.8.17-2) ...
Please run fvconfig generate to create and initialize the database
Setting up ca-certificates-java (20100412) ...
creating /etc/ssl/certs/java/cacerts...
  removed untrusted certificate mozilla/DigiNotar_Root_CA.crt
done.

[11:19:25] jbs@carlin:/home/jbs
+$ ls -Fla /usr/share/db/flowvisor /etc/flowvisor
/etc/flowvisor:
total 28
drwxr-xr-x  2 root root 4096 Apr 22 11:19 ./
drwxr-xr-x 56 root root 4096 Apr 22 11:19 ../
-rw-r--r--  1 root root 4682 Apr 19 19:17 FlowVisorDB.sql
-rw-r--r--  1 root root   98 Apr 19 19:17 derby.properties
-rw-r--r--  1 root root 1952 Apr 19 19:17 envs.sh
-rw-r--r--  1 root root  808 Apr 19 19:17 fvlog.config

/usr/share/db/flowvisor:
total 8
drwxr-xr-x 2 root root 4096 Apr 22 11:19 ./
drwxr-xr-x 3 root root 4096 Apr 22 11:19 ../
```

Did you try it? Did it do something different for you?

**Comment by leenheer** [comment]
Checked our build env and found the problem. Fixed in 0.8.17-3

**Comment by jbsbbn** [comment]
Tested, looks good. Thanks!

**Comment by leenheer** [comment]
-3 is .deb package revision. Problem was in .deb building process -
permissions of some config files were set to root iso flowvisor. .deb files
are on our deb/rpm repo not on github

marc

**Comment by alshabib** [comment]
You will also need to run sudo apr-get update to get the latest package.

## 

Ali Al-Shabibi (sent from handheld)

On 22 avr. 2013, at 09:57, leenheer notifications@github.com wrote:

-3 is .deb package revision. Problem was in .deb building process -
permissions of some config files were set to root iso flowvisor. .deb files
are on our deb/rpm repo not on github

marc
On Apr 22, 2013 9:54 AM, "mssumanth" notifications@github.com wrote:

> Hi Marc,
> 
> What was the problem over here?
> Could you please tell me?
> Also how do I get 0.8.17-3 from github, all I get is just 0.8.17 version
> of
> flowvisor!
> 
> Thanks
> Sumanth
> 
> On Mon, Apr 22, 2013 at 8:34 AM, leenheer notifications@github.com
> wrote:
> 
> > Checked our build env and found the problem. Fixed in 0.8.17-3
> > 
> > —
> > Reply to this email directly or view it on GitHub<
> 
> https://github.com/OPENNETWORKINGLAB/flowvisor/issues/199#issuecomment-16795208>
> 
> > .
> 
> —
> Reply to this email directly or view it on GitHub<
> https://github.com/OPENNETWORKINGLAB/flowvisor/issues/199#issuecomment-16801170>
> 
> .

—
Reply to this email directly or view it on
GitHubhttps://github.com/OPENNETWORKINGLAB/flowvisor/issues/199#issuecomment-16801461
.
